src/ValoremOptionsClearinghouse.sol

```diff
@@ -800,20 +808,19 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
                 unexercisedBucketIndices[exerciseIndex] = overwrite;
                 unexercisedBucketIndices.pop();
             } else {
-                // Bucket is partially exercised/assigned
+                // Bucket is partially exercised/assigned.
                 amountPresentlyExercised = amount;
                 amount = 0;
             }
             bucketInfo.amountExercised += amountPresentlyExercised;
 
+            emit BucketAssignedExercise(optionId, bucketIndex, amountPresentlyExercised);
```

Comment:
looks like this is the line that addresses offline accounting

Reply:
Right, but without the `BucketWrittenInto` event, an off-chain actor won't be able to make heads or tails of what this event means. They need to know how many options, for each claim/index, are being written into the buckets to support off-chain claim accounting.